source/initial_conditions/S40RTS_perturbation.cc

```diff
@@ -625,6 +631,10 @@ namespace aspect
                              Patterns::Double (0),
                              "The reference temperature that is perturbed by the spherical "
                              "harmonic functions. Only used in incompressible models.");
+          prm.declare_entry ("Remove temperature heterogeneity down to specified depth", "-10",
+                             Patterns::Double (),
+                             "This will set the heterogeneity prescribed by S20RTS or S40RTS to zero "
+                             "down to the specified depth.");
```

Comment:
I suspect that that can indeed happen. I think you probably want to set it to a very large negative value, e.g., `-std::numeric_limits<double>::max()` and its corresponding string representation which you can get via `boost::lexical_cast<std::string>(std::numeric_limits<double>::max())`.